azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,363 @@
+
+        # check if there is a spc called "opc-ua-connector", if not create one
+        try:
+            opcua_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        self._add_secret_to_spc(
+            secret_name=secret_name,
+            spc=opcua_spc,
+            resource_group=resource_group,
+            spc_keyvault_name=spc_keyvault_name,
+        )
+
+        return self._add_secret_to_secret_sync(
+            secret_name=secret_name,
+            file_name=file_name,
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+        )
+
+    def _process_fortos_yaml(self, object_text: str, secret_entry: Optional[dict] = None) -> str:
+        if object_text:
+            # TODO: formatting will be removed once fortos service fixes the formatting issue
+            object_text.replace("\n    - |", "\n- |")
+            objects_obj = yaml.safe_load(object_text)
+        else:
+            objects_obj = {"array": []}
+        entry_text = yaml.safe_dump(secret_entry, indent=6)
+        objects_obj["array"].append(entry_text)
+        object_text = yaml.safe_dump(objects_obj, indent=6)
+        return object_text.replace("\n- |", "\n    - |")
+
+    def _get_cl_resources(self, instance_name: str, resource_group: str) -> dict:
+        self.instance = self.instances.show(name=instance_name, resource_group_name=resource_group)
+        self.resource_map = self.instances.get_resource_map(self.instance)
+        custom_location = self.resource_client.resources.get_by_id(
+            resource_id=self.instance["extendedLocation"]["name"], api_version=CUSTOM_LOCATIONS_API_VERSION
+        )
+        cl_resources = self.resource_map.connected_cluster.get_aio_resources(custom_location_id=custom_location["id"])
+        return cl_resources
+
+    def _find_existing_spc(self, instance_name: str, cl_resources: List[dict]) -> dict:
+        secretsync_spc = None
+
+        if cl_resources:
+            for resource in cl_resources:
+                if resource["type"].lower() == "microsoft.secretsynccontroller/azurekeyvaultsecretproviderclasses":
+                    resource_id_container = parse_resource_id(resource["id"])
+                    secretsync_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                        resource_group_name=resource_id_container.resource_group_name,
+                        azure_key_vault_secret_provider_class_name=resource_id_container.resource_name,
+                    )
+                    break
+
+        if not secretsync_spc:
+            logger.error(
```

Comment:
logger.error or raise error?

Reply:
do we use raise error when user input is incorrect? in this case I'm not sure if secret sync not enable is a user error tho?